packages/report-server/src/reportBuilder/transform/parser/TransformParser.ts

```diff
@@ -71,14 +87,29 @@ export class TransformParser extends ExpressionParser {
     this.lookups.next = this.rows[this.currentRow + 1] || {};
     this.lookups.row = this.rows[this.currentRow];
     this.lookups.index = this.currentRow + 1;
-    this.set('$previous', this.lookups.previous);
-    this.set('$next', this.lookups.next);
-    this.set('$row', this.lookups.row);
-    this.set('$index', this.lookups.index);
+    this.set('_previous', this.lookups.previous);
+    this.set('_next', this.lookups.next);
+    this.set('_row', this.lookups.row);
+    this.set('_index', this.lookups.index);
     addRowToLookup(this.lookups.row, this.lookups.allPrevious);
+    this.addRowToScope(this.lookups.row);
   }
 
-  whereFunction = (check: (row: Row) => boolean) => {
+  public addRowToScope = (row: Row) => {
+    Object.entries(row).forEach(([field, value]) => {
+      if (value !== undefined && value !== null && !field.includes(' ')) {
```

Comment:
Hadn't really thought about this, but the shallow variable lookup will only work with columns that have no whitespace in them. There's always the `_row` lookup to get whitespace columns (eg. `_row['Total Cases']`)

Reply:
Yeah, I think this might be a tricky one to support. mathjs isn't liking the `'` in a variable name, and I'm worried if I change how it parses that char it'll break strings...

Reply:
Tricky, we can navigate that one later!